riot/runtime/import.ml

```diff
@@ -139,6 +139,35 @@ let spawn_pinned fn =
 
 let spawn_link fn = _spawn ~do_link:true fn
 
+let spawn_blocking fn =
+  let pool = _get_pool () in
+  (* Create a scheduler *)
+  let blocking_scheduler = Scheduler.Pool.spawn_blocking_scheduler pool in
+
+  (* Start the process *)
+  let proc =
+    Process.make blocking_scheduler.scheduler.uid (fun () ->
+        try
+          fn ();
+          Normal
+        with
+        | Proc_state.Unwind -> Normal
+        | exn ->
+            Log.error (fun f ->
+                f "Process %a died with unhandled exception %s:\n%s" Pid.pp
+                  (self ()) (Printexc.to_string exn)
+                  (Printexc.get_backtrace ()));
+
+            Exception exn)
+  in
+  Process.set_flag proc (IsBlockingProc true);
```

Comment:
Here is we basically say "when this processes exits, we can go ahead and shutdown the scheduler".
The main thing here is I don't want to modify the raw Process and Scheduler code too much making a bunch of duplicated functions.
I struggled to figure out how to do that from the scheduler's point of view rather than the process's, as there's no way to determine in the `run` / `run_loop` whether it *should* exit after a process is finished, hence I added the `mutable stop : bool` flag.